src/node_port.py

```diff
@@ -106,3 +124,28 @@ def apply_service(self, service: Service) -> None:
                 return
             else:
                 raise
+
+    def delete_unit_service(self) -> None:
+        """Deletes a unit Service, if it exists."""
+        try:
+            service = self.get_unit_service()
+        except ApiError as e:
+            if e.status.code == 404:
+                logger.debug(
+                    f"Could not find {self.get_unit_service_name()} to delete."
+                )
+                return
+
+        if not service.metadata:
+            raise Exception(f"Could not find metadata for {service}")
+
+        try:
+            self.client.delete(Service, service.metadata.name)
+        except ApiError as e:
+            if e.status.code == 403:
+                logger.error("Could not delete service, application needs `juju trust`")
```

Comment:
Should this not raise as well? i wonder if logging is enough or should it rather be set on the status?

Reply:
Handled [in following pr](https://github.com/canonical/mongos-k8s-operator/pull/32/files#diff-fc7286c5c4de1116862878bad8c3293dd1d525f3eb87642f92e66b48732dc298)